inc/hclib-async.h

```diff
@@ -64,7 +64,9 @@ inline void call_lambda(T* lambda) {
 	const int wid = current_ws()->id;
 	MARK_BUSY(wid);
 	(*lambda)();
-	HC_FREE((void*) lambda);
+    // FIXME - this whole call chain is kind of a mess
```

Comment:
Nick, maintaining backward compatibility with HCUPC++ might not be easy as hclib codebase has significantly changed (e.g. continuation creation has broken the support of old design of communication worker). Max is working a generic resource worker that he plans as a pluggable module for any communication library (e.g., MPI, UPC++, OpenSHMEM) and I believe he would merge that branch into master soon.